samples/grids/data-grid/load-save-layout/src/index.tsx

```diff
@@ -0,0 +1,378 @@
+            gaugeValue = content.children[1] as HTMLSpanElement;
+        }
+
+        // conditional formatting:
+        if (sales < 400000) {
+            gaugeValue.style.color = "rgb(211, 17, 3)";
+            gaugeBar.style.background = "rgb(211, 17, 3)";
+        }
+        else if (sales < 650000) {
+            gaugeValue.style.color = "Orange";
+            gaugeBar.style.background = "Orange";
+        }
+        else {
+            gaugeValue.style.color = "rgb(21, 190, 6)";
+            gaugeBar.style.background = "rgb(21, 190, 6)";
+        }
+        gaugeValue.textContent = "$" + sales / 1000 + ",000";
+
+        let gaugeWidth = (sales / 990000) * 100;
+        gaugeWidth = Math.min(100, gaugeWidth);
+        gaugeBar.style.width = gaugeWidth + "%";
+    }
+
+    public onEmailCellUpdating(s: IgrTemplateColumn, e: IgrTemplateCellUpdatingEventArgs) {
+        const content = e.content as HTMLDivElement;
+        const info = e.cellInfo as IgrTemplateCellInfo;
+        const item = info.rowItem;
+        let link: HTMLAnchorElement;
+
+        if (content.childElementCount === 0) {
+            link = document.createElement("a");
+
+            content.style.verticalAlign = "center";
+            content.style.marginTop = "15px";
+            content.style.lineHeight = "normal";
+            content.style.display = "inline-grid";
+            content.style.fontFamily = "Verdana";
+            content.style.fontSize = "13px";
+            content.style.color = "#4286f4";
+            content.style.width = "100%";
+
+            content.appendChild(link);
+        } else {
+            link = content.children[0] as HTMLAnchorElement;
+        }
+
+        link.href = "mailto:" + item.Email + "?Subject=Hello%20Friend";
+        link.textContent = item.Email;
+    }
+
+    public onPhoneCellUpdating(s: IgrTemplateColumn, e: IgrTemplateCellUpdatingEventArgs) {
+        const content = e.content as HTMLDivElement;
+
+        const info = e.cellInfo as IgrTemplateCellInfo;
+        const item = info.rowItem;
+        let link: HTMLAnchorElement;
+
+        if (content.childElementCount === 0) {
+
+            link = document.createElement("a");
+
+            content.style.verticalAlign = "center";
+            content.style.marginTop = "15px";
+            content.style.lineHeight = "normal";
+            content.style.display = "inline-block";
+            // content.style.display = "inline-grid";
+            content.style.fontFamily = "Verdana";
+            content.style.fontSize = "13px";
+            content.style.color = "#4286f4";
+            content.style.width = "100%";
+
+            content.appendChild(link);
+        } else {
+            link = content.children[0] as HTMLAnchorElement;
+        }
+
+        link.href = "tel:" + item.Phone;
+        link.textContent = item.Phone;
+    }
+
+    public onLoadLayoutClicked() {
+        this.grid.loadLayout(this.savedLayout);
+    }
+
+    public onSaveLayoutClicked() {
```

Comment:
move this function above render() so it is easier to find in code-viewer

Reply:
resolved